src/rgw/rgw_tag_s3.cc

```diff
@@ -0,0 +1,82 @@
+#include <map>
+#include <string>
+#include <iostream>
+
+#include "include/types.h"
+
+#include "rgw_tag_s3.h"
+
+using namespace std;
+
+bool RGWObjTagEntry_S3::xml_end(const char* el){
+  RGWObjTagKey_S3 *key_obj = static_cast<RGWObjTagKey_S3 *>(find_first("Key"));
+  RGWObjTagValue_S3 *val_obj = static_cast<RGWObjTagValue_S3 *>(find_first("Value"));
+
+  if (!key_obj)
+    return false;
+
+  string s = key_obj->get_data();
+  if (s.empty()){
+    return false;
+  }
+
+  key = s;
+  if (val_obj) {
+    val = val_obj->get_data();
+  }
+
+  return true;
+}
+
+bool RGWObjTagSet_S3::xml_end(const char* el){
```

Comment:
If we don't actually use a parameter (but need it to match a type signature) it's probably best to leave it unnamed so it's explicit.